lib/std/Thread/Mutex.zig

```diff
@@ -50,6 +50,13 @@ pub fn unlock(self: *Mutex) void {
     self.impl.unlock();
 }
 
+/// Asserts that the mutex was previously acquired with `lock()` or `tryLock()`.
```

Comment:
Having it be a no-op outside debug/ST seems like a correctness issue. Not sure if something like this should be exposed.

Reply:
I could add my `std.debug.assert(!self.mutex.tryLock());` variant in ReleaseSafe,
but I guess that would again come with the same problems.

An alternative would be to extend the `DebugImpl` to ReleaseSafe.

Reply:
my suggestion would be to have it run in all modes

Reply:
In general, assertions don't run in release mode. It seems entirely reasonable for it to not try to lock the mutex in release mode, this assertion not locking the mutex in release mode is no more a correctness issue than `std.Arraylist(...).appendAssumeCapacity` not crashing if the capacity is exceeded in release mode.

Reply:
Assertions are kept in safe modes rather than debug mode specifically; It should be enabled in ReleaseSafe but I assume that would mean adding the `Thread.Id` checks to that mode too which sounds undesirable.

Reply:
Yeah you are right, and there doesn't seem to be a good solution for this that makes everyone happy.
I guess I'll close this PR then and implement it in my own code, sorry for bothering you all.